apps/ledger-live-desktop/src/newArch/features/WalletSync/__tests__/useAddMember.test.ts

```diff
@@ -0,0 +1,173 @@
+  });
+
+  it("should handle missing device", async () => {
+    const { result } = renderHook(() => useAddMember({ device: null }));
+
+    expect(result.current.error).toBeNull();
+    expect(Mocks.setTrustchain).not.toHaveBeenCalledWith(trustchain);
+    expect(Mocks.setFlow).toHaveBeenCalledWith({
+      flow: Flow.Activation,
+      step: Step.DeviceAction,
+    });
+    expect(Mocks.track).not.toHaveBeenCalled();
+  });
+
+  it("should handle already initialized trustchain", async () => {
+    Mocks.sdk.getOrCreateTrustchain.mockRejectedValue(new TrustchainAlreadyInitialized());
+    const { result } = renderHook(() => useAddMember({ device }));
+
+    await waitFor(() => expect(Mocks.sdk.getOrCreateTrustchain).toHaveBeenCalled());
+
+    expect(result.current.error).toBeNull();
+    expect(Mocks.setTrustchain).not.toHaveBeenCalledWith(trustchain);
+    expect(Mocks.setFlow).toHaveBeenCalledWith({
+      flow: Flow.Synchronize,
+      step: Step.AlreadySecuredSameSeed,
+    });
+    expect(Mocks.track).not.toHaveBeenCalled();
+  });
+
+  it("should handle trustchain initialized with other seed", async () => {
+    Mocks.sdk.getOrCreateTrustchain.mockRejectedValue(
+      new TrustchainAlreadyInitializedWithOtherSeed(),
+    );
+    const { result } = renderHook(() => useAddMember({ device }));
+
+    await waitFor(() => expect(Mocks.sdk.getOrCreateTrustchain).toHaveBeenCalled());
+
+    expect(result.current.error).toBeNull();
+    expect(Mocks.setTrustchain).not.toHaveBeenCalledWith(trustchain);
+    expect(Mocks.setFlow).toHaveBeenCalledWith({
+      flow: Flow.Synchronize,
+      step: Step.AlreadySecuredOtherSeed,
+    });
+    expect(Mocks.track).not.toHaveBeenCalled();
+  });
+
+  it("should handle missing member credentials", async () => {
+    Mocks.memberCredentialsSelector.mockReturnValue(null);
+    const { result } = renderHook(() => useAddMember({ device }));
+
+    expect(result.current.error).toBeInstanceOf(Error);
+    expect(Mocks.sdk.getOrCreateTrustchain).not.toHaveBeenCalled();
+    expect(Mocks.setTrustchain).not.toHaveBeenCalledWith(trustchain);
+    expect(Mocks.setFlow).not.toHaveBeenCalled();
+    expect(Mocks.track).not.toHaveBeenCalled();
+  });
+
+  it("should handle other sdk errors", async () => {
+    Mocks.sdk.getOrCreateTrustchain.mockRejectedValue(new Error("Random error"));
+    const { result } = renderHook(() => useAddMember({ device }));
+
+    await waitFor(() => expect(Mocks.sdk.getOrCreateTrustchain).toHaveBeenCalled());
+
+    expect(result.current.error).toBeInstanceOf(Error);
+    expect((result.current.error as Error).message).toBe("Random error");
+    expect(Mocks.setTrustchain).not.toHaveBeenCalled();
+    expect(Mocks.setFlow).not.toHaveBeenCalled();
+    expect(Mocks.track).not.toHaveBeenCalled();
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    Mocks.memberCredentialsSelector.mockReturnValue({
+      pubkey: "pubkey",
+      privatekey: "privatekey",
+    });
+  });
+});
+
```

Comment:
i would move this bloc to the top before using the variables

Reply:
Do you mean the beforeEach ?